make/libraries

```diff
@@ -87,7 +87,9 @@ $(STAN_SUNDIALS_HEADERS) : $(LIBSUNDIALS)
 clean-sundials:
 	@echo '  cleaning sundials targets'
 	$(RM) $(wildcard $(sort $(SUNDIALS_CVODES) $(SUNDIALS_IDAS) $(SUNDIALS_KINSOL) $(SUNDIALS_NVECSERIAL) $(LIBSUNDIALS)))
-
+else
+LIBSUNDIALS := $(SUNDIALS_LIB)/libsundials_nvecserial.a $(SUNDIALS_LIB)/libsundials_cvodes.a $(SUNDIALS_LIB)/libsundials_idas.a $(SUNDIALS_LIB)/libsundials_kinsol.a
```

Comment:
This didn't work for me locally when I tried, since my installation of sundials was build dynamically (.so) rather than statically (.a).

If instead of passing e.g. `$(SUNDIALS_LIB)/libsundials_nvecserial.a ` on the command line I instead passed `-lsundials_nvecserial` it worked, and this should work either way. I believe that this is how we do TBB. I was able to do this by changing this like to 
```makefile
LIBSUNDIALS := -lsundials_nvecserial -lsundials_cvodes -lsundials_idas -lsundials_kinsol
```

I also needed to update the likes in make/test that look like
```makefile
$(ALGEBRA_SOLVER_TESTS) : $(LIBSUNDIALS)
```
to instead be 
```makefile
$(ALGEBRA_SOLVER_TESTS) : LDLIBS += $(LIBSUNDIALS)
```


After this I could build the algebra solver tests with a sundials installed from conda. A few of them printed errors (`[KINSOL ERROR]  KINSol  The linear solver's setup function failed in an unrecoverable manner.`) but they at least claimed to have passed.

Reply:
Reading the test now, those errors are expected. So it's just the linking issue that needs to be resolved.

Now that I have a working full-conda example on Linux I'm excited to try it out on Windows.

Reply:
This should probably be split up into a `SUNDIALS_TARGETS` and `LDLIBS_SUNDIALS` like tbb is, since we don't want `make -f make/standalone math-libs` to try to build "-lsundials_idas" etc. 

Right now `LIBSUNDIALS` serves as both a build target and a set of objects to link against, which isn't going to fly for this kind of thing